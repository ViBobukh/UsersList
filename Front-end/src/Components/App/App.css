body {
    font-family: Bahnschrift, sens-serif;
    font-size: large;
}

input {
    font-family: Bahnschrift, sens-serif;
    font-size: large;
}