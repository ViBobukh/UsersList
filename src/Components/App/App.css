.app-btn {
    background-color: #28b07b;
    border-radius: 6px;
    border: 2px solid black;
    width: 20%;
    height: 8vh;
    font-size: 1.8vw;
    font-family: Bahnschrift, sans-serif;
    margin-left: 79.8%;
}