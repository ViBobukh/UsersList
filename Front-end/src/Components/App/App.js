import React, { useCallback, useEffect, useMemo, useState } from "react";
import "./App.css";
import List from "../List/List";
import { filter } from "../../lib/filter";
import { getUsers } from "../../lib/getUsers";

/**
 * Main components with input and filter
 * @returns {JSX.Element}
 */
function App() {
  const [allUsers, setUsers] = useState([]);
  const [titleName, setTitleName] = useState("");
  const [titleEmail, setTitleEmail] = useState("");
  const [currentInput, setCurrentInput] = useState("");

  /**
   * Get all users from JSON
   */
  useEffect(() => {
    getUsers().then((res) => setUsers(res));
  }, []);

  /**
   * Callback for setting title by text of input
   * @type {function(*): void}
   */
  const searching = useCallback(
    (event) => {
      switch (event.currentTarget.name) {
        case "name":
          setCurrentInput("name");
          setTitleName(event.currentTarget.value);
          break;
        case "email":
          setCurrentInput("email");
          setTitleEmail(event.currentTarget.value);
          break;
      }
    },
    [titleName, titleEmail]
  );

  /**
   * ReactHook with changed array users
   * @type {Array}
   */
  const filteredUsers = useMemo(() => {
    switch (currentInput) {
      case "name":
        return filter(allUsers, titleName, currentInput);
      case "email":
        return filter(allUsers, titleEmail, currentInput);
    }
  }, [titleName, titleEmail, allUsers]);

  return (
    <div>
      <input
        onChange={searching}
        value={titleName}
        name="name"
        placeholder="Name searching"
      />
      <input
        name="email"
        onChange={searching}
        value={titleEmail}
        placeholder="Email searching"
      />
      <List users={titleName || titleEmail ? filteredUsers : allUsers} />
    </div>
  );
}

export default App;
