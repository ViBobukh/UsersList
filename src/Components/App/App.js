import React from "react";
import './App.css';

function App (){

    return (
        <div>
            <p>aaaa</p>
        </div>
    );
}

export default App;