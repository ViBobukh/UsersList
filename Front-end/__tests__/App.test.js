import React from "react";
import renderer from "react-test-renderer";

import App from "../src/Components/App/App";

describe("App test", () => {
  it("App render", () => {
    const component = renderer.create(<App />);
    let domTree = component.toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
