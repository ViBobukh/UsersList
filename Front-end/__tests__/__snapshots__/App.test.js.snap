// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test App render 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name searching"
    value=""
  />
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email searching"
    value=""
  />
  <ul>
    <li>
      <div
        className="user"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Maks
        </div>
      </div>
    </li>
    <li>
      <div
        className="user"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Mark
        </div>
      </div>
    </li>
    <li>
      <div
        className="user"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Mario
        </div>
      </div>
    </li>
    <li>
      <div
        className="user"
        onClick={[Function]}
      >
        <div
          className=""
        >
          Patrick
        </div>
      </div>
    </li>
  </ul>
</div>
`;
